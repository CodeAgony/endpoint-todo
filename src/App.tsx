import { createGlobalStyle } from "styled-components";
import { Header } from "./components/Header";
import { TodoList } from "./components/TodoList";

const AppStyle = createGlobalStyle`
  body {
    font-family: sans-serif, serif;
  }
`;

function App() {
  return (
    <div data-testid="app-root">
      <Header />
      <TodoList />
      <AppStyle />
    </div>
  );
}

export default App;
