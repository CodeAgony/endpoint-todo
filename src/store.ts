import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import rootReducer from "./reducers";

const store = createStore(rootReducer, applyMiddleware(thunk));

type AppDispatch = typeof store.dispatch;

export type RootState = ReturnType<typeof store.getState>;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export default store;
