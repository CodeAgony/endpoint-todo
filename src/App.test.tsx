import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import * as reduxHooks from "./store";
import App from "./App";
import todosMock from "./mocks/fetchTodos";

describe("App", () => {
  beforeEach(() => {
    jest.spyOn(reduxHooks, "useAppSelector").mockReturnValue({ isLoading: false, todos: todosMock });
  });

  const initialState = { isLoading: false, todos: todosMock };
  const mockStore = configureStore([thunk]);
  const store = mockStore(initialState);

  const app = (
    <Provider store={store}>
      <App />
    </Provider>
  );

  test("renders", () => {
    render(app);
    expect(screen.getByTestId("app-root")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const tree = renderer.create(app).toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
