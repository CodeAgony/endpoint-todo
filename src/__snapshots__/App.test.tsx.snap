// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  data-testid="app-root"
>
  <header
    className="sc-bdnxRM jnVsTK"
    data-testid="header"
  >
    <div
      className="sc-gtsrHT jWdEKn"
    >
      Todo App
    </div>
  </header>
  <main
    className="sc-pNWdM hgPrSl"
    data-testid="todo-list"
  >
    <div
      className="sc-gKAaRy eruBWu"
      data-testid="todo-item"
    >
      <label
        className="sc-hKFxyN kxtYU sc-fujyAs jbNzwC"
        data-testid="checkbox"
      >
        <input
          checked={true}
          className="sc-eCApnc kCTsEE"
          onChange={[Function]}
          readOnly={false}
          type="checkbox"
        />
        <div
          className="sc-jSFjdj ibnWIs"
          data-testid="checkbox-label"
        >
          File 2020 Taxes
        </div>
      </label>
      <time
        className="sc-iCoGMd iqsebk"
        data-tesid="todo-due-date"
      >
        03/10/2020
      </time>
    </div>
    <div
      className="sc-gKAaRy eruBWu"
      data-testid="todo-item"
    >
      <label
        className="sc-hKFxyN kxtYU sc-fujyAs jbNzwC"
        data-testid="checkbox"
      >
        <input
          checked={true}
          className="sc-eCApnc kCTsEE"
          onChange={[Function]}
          readOnly={false}
          type="checkbox"
        />
        <div
          className="sc-jSFjdj ibnWIs"
          data-testid="checkbox-label"
        >
          Fold laundry
        </div>
      </label>
    </div>
    <div
      className="sc-gKAaRy klnoye"
      data-testid="todo-item"
    >
      <label
        className="sc-hKFxyN kxtYU sc-fujyAs wDCNW"
        data-testid="checkbox"
      >
        <input
          checked={false}
          className="sc-eCApnc kCTsEE"
          onChange={[Function]}
          readOnly={false}
          type="checkbox"
        />
        <div
          className="sc-jSFjdj ibnWIs"
          data-testid="checkbox-label"
        >
          Call Mom
        </div>
      </label>
      <time
        className="sc-iCoGMd iqsebk"
        data-tesid="todo-due-date"
      >
        06/26/2020
      </time>
    </div>
    <div
      className="sc-gKAaRy euYuUz"
      data-testid="todo-item"
    >
      <label
        className="sc-hKFxyN kxtYU sc-fujyAs wDCNW"
        data-testid="checkbox"
      >
        <input
          checked={false}
          className="sc-eCApnc kCTsEE"
          onChange={[Function]}
          readOnly={false}
          type="checkbox"
        />
        <div
          className="sc-jSFjdj ibnWIs"
          data-testid="checkbox-label"
        >
          Walk the dog
        </div>
      </label>
    </div>
    <div
      className="sc-gKAaRy klnoye"
      data-testid="todo-item"
    >
      <label
        className="sc-hKFxyN kxtYU sc-fujyAs wDCNW"
        data-testid="checkbox"
      >
        <input
          checked={false}
          className="sc-eCApnc kCTsEE"
          onChange={[Function]}
          readOnly={false}
          type="checkbox"
        />
        <div
          className="sc-jSFjdj ibnWIs"
          data-testid="checkbox-label"
        >
          Feed the cat
        </div>
      </label>
      <time
        className="sc-iCoGMd iqsebk"
        data-tesid="todo-due-date"
      >
        06/24/2020
      </time>
    </div>
    <div
      className="sc-gKAaRy klnoye"
      data-testid="todo-item"
    >
      <label
        className="sc-hKFxyN kxtYU sc-fujyAs wDCNW"
        data-testid="checkbox"
      >
        <input
          checked={false}
          className="sc-eCApnc kCTsEE"
          onChange={[Function]}
          readOnly={false}
          type="checkbox"
        />
        <div
          className="sc-jSFjdj ibnWIs"
          data-testid="checkbox-label"
        >
          Run LA marathon
        </div>
      </label>
      <time
        className="sc-iCoGMd iqsebk"
        data-tesid="todo-due-date"
      >
        03/21/2021
      </time>
    </div>
  </main>
</div>
`;
